brainframe/cli/docker_compose.py

```diff
@@ -57,58 +66,48 @@ def run(install_path: Path, commands: List[str]) -> None:
 def download(target: Path, version: str = "latest") -> None:
     _assert_has_write_permissions(target.parent)
 
-    subdomain, auth_flags, version = check_download_version(version=version)
+    if version == "latest":
+        version = get_latest_version()
+
+    credentials = config.staging_credentials()
 
     url = BRAINFRAME_DOCKER_COMPOSE_URL.format(
-        subdomain=subdomain, version=version
-    )
-    os_utils.run(
-        ["curl", "-o", str(target), "--fail", "--location", url] + auth_flags,
+        subdomain="staging." if config.is_staging.value else "",
+        version=version,
     )
+    response = requests.get(url, auth=credentials, stream=True)
```

Comment:
I was wondering why you do stream=True in this codebase. Is there some benefit to not reading the whole response immediately?

Reply:
Using `stream=True` prevents the entire file from being put into memory at once. This is more important for the CLI binary when self-updating than the `docker-compose.yml` admittedly, but adding `stream=True` didn't complicate usage here and I would argue is somewhat better practice.

Reply:
Right, that makes sense. Though the moment you do `response.text` it will copy it all. Either way, no big deal!